### PR TITLE
Add validator for basic validations

### PR DESCRIPTION
**Pull request by kasunfdo**
Changes proposed in this pull request:
- Add validator for basic validations

### CODE REVIEW

File: sysl2/sysl/validate.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		}
+	} else {
+		return content{
+			name:    stmt.GetAssign().GetName(),
+			typeRef: typeRef(stmt.GetAssign().GetExpr()),
+		}
+	}
+}
+
+func returnType(syslType *sysl.Type) string {
```
Comment (sahejsingh) [inline sysl2/sysl/validate.go]: Returntype is probably not the right name for it

File: sysl2/sysl/codegen.go [hunk 2]
```diff
@@ -231,10 +231,14 @@ func GenerateCode(root_model, model, root_transform, transform, grammar, start s
 
 	mod, modelAppName := loadAndGetDefaultApp(root_model, model)
 	tx, transformAppName := loadAndGetDefaultApp(root_transform, transform)
+	grm, _ := loadAndGetDefaultApp("grammars", "go.gen.sysl")
```
Comment (sahejsingh) [inline sysl2/sysl/codegen.go]: How will it work with other grammar models?

File: sysl2/sysl/validate.go (hunk 1 above)
Comment (kasunfdo) [reply sysl2/sysl/validate.go]: Changed

File: sysl2/sysl/codegen.go (hunk 2 above)
Comment (kasunfdo) [reply sysl2/sysl/codegen.go]: My bad. I fixed it.

File: sysl2/sysl/validate.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+func typeRef(expr *sysl.Expr) string {
+	if expr.GetLiteral() != nil {
+		return "string"
+	}
+	if expr.GetList() != nil {
+		return "list"
+	}
+	if call := expr.GetCall(); call != nil {
+		if _, exists := GoFuncMap[call.GetFunc()]; exists {
+			return "string"
```
Comment (sahejsingh) [inline sysl2/sysl/validate.go]: Not all calls return `string`, some return `array of string`

File: sysl2/sysl/validate.go [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+type ValidationMsg struct {
+	message string
+	msgType int
+}
+
+type Views map[string]View
+
+func typeRef(expr *sysl.Expr) string {
+	if expr.GetLiteral() != nil {
```
Comment (sahejsingh) [inline sysl2/sysl/validate.go]: why not reuse the existing way to represent type in sysl i.e `sysl.Type`?

File: sysl2/sysl/validate_test.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+import (
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"testing"
+)
+
+func TestTypeRef(t *testing.T) {
+	transform, _ := loadAndGetDefaultApp("tests", "validation_transform.sysl")
+
+	typeRefs := []string{"list", "string", "BinExpr", "Conditional", "string", "View", "Transform", "variable"}
```
Comment (sahejsingh) [inline sysl2/sysl/validate_test.go]: only string is a type here. Non of the other things mentioned are types!

File: sysl2/sysl/codegen.go [hunk 6]
```diff
@@ -2,16 +2,15 @@ package main
...
 	"github.com/anz-bank/sysl/src/proto"
-	parser "github.com/anz-bank/sysl/sysl2/naive"
+	"github.com/anz-bank/sysl/sysl2/naive"
 	ebnfGrammar "github.com/anz-bank/sysl/sysl2/proto"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"os"
+	"sort"
```
[the remark this answers is not in the document]
Comment (kasunfdo) [reply sysl2/sysl/codegen.go]: gofmt did the change.

[the remark this answers is not in the document]
Comment (marcelocantos) [reply sysl2/sysl/codegen.go]: That would only happen if you removed the empty line between the built-in and third-party packages.

**Pull request closed without merging**